tests/unit/s2n_security_policies_test.c

```diff
@@ -312,41 +312,38 @@ int main(int argc, char **argv)
         struct s2n_connection *conn = s2n_connection_new(S2N_CLIENT);
         s2n_connection_set_config(conn, config);
 
-        EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "default"));
+        EXPECT_SUCCESS(s2n_connection_set_security_policy(conn, S2N_SECURITY_POLICY_DEFAULT));
         EXPECT_SUCCESS(s2n_connection_get_security_policy(conn, &security_policy));
```

Comment:
https://github.com/awslabs/s2n/pull/2086#discussion_r520905452 Where is s2n_connection_set_cipher_preferences still being tested? And do we test s2n_config_set_security_policy anywhere?

Reply:
To set context, the only reason `s2n_connection_set_cipher_preferences` was used is because we didn't have the security policy setter yet. This module is meant to test security policies, not cipher preferences.

`s2n_connection_set_cipher_preferences` is still being tested in the same locations prior to the security policies being added. If that testing is insufficient, we can open an issue to write tests.

`s2n_config_set_security_policy` was missing. To add those tests,  I've modified the `s2n_config_set_cipher_preferences` test in the same way I modified `s2n_connection_set_cipher_preferences`.

Reply:
"s2n_connection_set_cipher_preferences is still being tested in the same locations prior to the security policies being added" -- where are those places?

Even if we didn't test set_cipher_preferences pre-security-policies, we do test it now. We probably shouldn't remove any test coverage as part of this change.

Reply:
I wouldn't block this PR on the test coverage of set_cipher_preferences. @ttjsu-aws  can you talk about why you used set_ciper_preferences in a security policy test and what a good course of action is here?

Reply:
Why not just duplicate the tests for the new API instead of removing them from the old API? Why do we need to remove existing coverage in order to cover the new API?

Reply:
Sure, duplicated the tests so we keep both sets.